### PR TITLE
DTS Watchdog cleanups

### PR DESCRIPTION
**Pull request by galak**
Now that all watchdog drivers are DTS aware we can make some clean ups associated with using DTS defines everywhere and removing CONFIG_WDT_0_NAME.

### CODE REVIEW

File: boards/arm/mps2_an385/mps2_an385.dts [hunk 1]
```diff
@@ -15,6 +15,7 @@
 		led1 = &led_1;
 		sw0 = &user_button_0;
 		sw1 = &user_button_1;
+		watchdog-0 = &wdog0;
```
Comment (erwango) [inline boards/arm/mps2_an385/mps2_an385.dts]: Would be nice to have consistency between zephyr aliases, so 'watchdog0'

File: drivers/watchdog/Kconfig [hunk 2]
```diff
@@ -8,6 +8,7 @@
 
 menuconfig WATCHDOG
 	bool "Watchdog Support"
+	select HAS_DTS_WDT
```
Comment (erwango) [inline drivers/watchdog/Kconfig]: Can't we simply get rid of it?

**Review by erwango — CHANGES_REQUESTED** [review]

File: drivers/watchdog/Kconfig (hunk 2 above)
Comment (galak) [reply drivers/watchdog/Kconfig]: We can, I wanted to keep it around until after the release.

File: boards/arm/mps2_an385/mps2_an385.dts (hunk 1 above)
Comment (galak) [reply boards/arm/mps2_an385/mps2_an385.dts]: Yeah, I wasn't sure about this one.  There are aliases for some SoCs that are of the form `<FOO>-<n>`.

Comment (galak) [reply boards/arm/mps2_an385/mps2_an385.dts]: Look at dts/arm/nxp/nxp_k8x.dtsi

Comment (erwango) [reply boards/arm/mps2_an385/mps2_an385.dts]: Indeed, but these are not zephyr aliases (as the ones that can be used in samples).
I don't know what they're needed for, but looks like vendor specifics, and possibly reuse from Linux (as I've seen such there as well).

Comment (galak) [reply boards/arm/mps2_an385/mps2_an385.dts]: Changed to `watchdog0`

Comment (galak) [reply boards/arm/mps2_an385/mps2_an385.dts]: Now I'm torn, we seem to have a mix of with and without the `-` used for samples & tests aliases.  See PR #20334

**Review by erwango — CHANGES_REQUESTED** [review]

File: boards/arm/arduino_due/arduino_due.dts [hunk 3]
```diff
@@ -9,7 +9,6 @@
 	compatible = "arduino,due", "atmel,sam3x8e", "atmel,sam3x";
 
 	aliases {
-		wdog = &wdog;
```
[the remark this answers is not in the document]
Comment (galak) [reply boards/arm/arduino_due/arduino_due.dts]: Replace it, the new `watchdog0` alias gets pulled in via the sam*.dtsi files.

File: boards/arm/sam4s_xplained/sam4s_xplained.dts [hunk 4]
```diff
@@ -17,7 +17,6 @@
 		led0 = &yellow_led_1;
 		led1 = &yellow_led_2;
 		sw0 = &user_button;
-		wdog = &wdog;
```
[the remark this answers is not in the document]
Comment (galak) [reply boards/arm/sam4s_xplained/sam4s_xplained.dts]: Replace it, the new `watchdog0` alias gets pulled in via the sam*.dtsi files.

File: boards/arm/sam_e70_xplained/sam_e70_xplained.dts [hunk 5]
```diff
@@ -19,7 +19,6 @@
 		i2c-2 = &i2c2;
 		led0 = &green_led;
 		sw0 = &sw0_user_button;
-		wdog = &wdog;
```
[the remark this answers is not in the document]
Comment (galak) [reply boards/arm/sam_e70_xplained/sam_e70_xplained.dts]: Replace it, the new `watchdog0` alias gets pulled in via the sam*.dtsi files.

**Review by erwango — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**